src/subject/commands/cancel-subject-enrollment-command.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/database/prisma.service';
+import { SubjectService } from '../subject.service';
+import { StudentNotEnrolledException } from '../utils/exceptions';
+
+@Injectable()
+export class CancelSubjectEnrollmentCommand {
+  constructor(
+    private readonly prisma: PrismaService,
+    private readonly subjectService: SubjectService,
+  ) {}
+
+  async execute(subjectId: number, studentId: number): Promise<void> {
+    const subject = await this.subjectService.findSubjectById(
+      studentId,
+      subjectId,
+    );
+
+    const enrollment = subject.studentsEnrolled.filter(
+      (enrollment) =>
+        enrollment.student_id === studentId &&
+        enrollment.subject_id === subjectId,
+    )[0];
```

Comment:
É menos custoso buscar no banco diretamente pela matrícula do que fazer essa iteração:

```suggestion
    const enrollment =
      await this.prismaService.subjectEnrollment.findFirst({
        where: {
          student_id: user_id,
          subject_id: data.subject_id,
          canceled_at: null,
        },
      });
```